### PR TITLE
Fixes issue where Alert crashes

### PR DESCRIPTION
**Pull request by rozele**
### Type of Change
_Erase all that don't apply._
- Bug fix (non-breaking change which fixes an issue)

### Why
There are edge cases where an Alert may be shown while the app is exiting. Generally, when the app is exiting, apps will have set the XamlRoot instance on XamlUIService to null. In this case, we may still attempt to show the alert on a XAML Island with a null XamlRoot instance, which would cause a crash.

### What
This change just ensures that, if we are in a XAML Island and we don't have a valid XamlRoot to attach to the ContentDialog, we just treat the alert as dismissed.

## Testing
Set up simple.tsx example to wire up onDismiss, added native button to Playground-Win32 to clear XamlRoot setting on XamlUIService, confirmed that trying to show an alert after setting XamlRoot to null on XamlUIService results in `onDismiss` callback.

 ###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://api.github.com/repos/microsoft/react-native-windows/pulls/11168)

### CODE REVIEW

**Comment by acoates-ms** [comment]
My only worry is if someone hasn't set the XamlRoot correctly in the normal case, their alerts will just silently fail.  -- Any ideas?

**Comment by rozele** [comment]
> My only worry is if someone hasn't set the XamlRoot correctly in the normal case, their alerts will just silently fail. -- Any ideas?

YellowBox warning for debug builds?

File: vnext/Microsoft.ReactNative/Modules/AlertModule.cpp [hunk 1]
```diff
@@ -98,6 +98,13 @@ void Alert::ProcessPendingAlertRequests() noexcept {
       });
 
       dialog.Closed([=](auto &&, auto &&) { xamlRoot.Changed(rootChangedToken); });
+    } else if (IsXamlIsland()) {
+      // We cannot show a ContentDialog in a XAML Island unless it is assigned a
+      // XamlRoot instance. In such cases, we just treat the alert as dismissed.
+      jsDispatcher.Post([result, this] { result(m_constants.dismissed, m_constants.buttonNeutral); });
+      pendingAlerts.pop();
+      ProcessPendingAlertRequests();
+      return;
```
Comment (jonthysell) [inline vnext/Microsoft.ReactNative/Modules/AlertModule.cpp:107]: Where does the crash actually occur? This `return` prevents the code from getting further, but I don't see how this behavior change is any different for people who call this code early without having set xamlroot as @acoates-ms indicated.

Comment (rozele) [reply vnext/Microsoft.ReactNative/Modules/AlertModule.cpp:107]: Crashes here: https://github.com/microsoft/react-native-windows/pull/11168/files#diff-3c81c385c361994bfc625387139150a42c69fc47f4b141648b55a327b8c54dacR128

Basically, you must set a XamlRoot before trying to show any alerts.

We could leave them queued indefinitely, but that seems like a race /  hang waiting to happen. E.g., imagine an app that creates a Promise from the `Alert.alert` call that gets resolved when a button is pressed or rejected when canceled or dismissed. Such apps would hang indefinitely.

Comment (acoates-ms) [reply vnext/Microsoft.ReactNative/Modules/AlertModule.cpp:107]: As an alternative, would it make sense to take the implementation I added in https://github.com/microsoft/react-native-windows/pull/11218 and fallback to that if there isn't a XamlRoot?

Comment (rozele) [reply vnext/Microsoft.ReactNative/Modules/AlertModule.cpp:107]: @acoates-ms wouldn't that just run into the same issue when the top-level HWND is not set?

**Comment by chiaramooney** [comment]
ping @acoates-ms

**Review by acoates-ms — APPROVED** [review]

**Pull request merged**